### ISSUE TITLE
Surprised that only the first predicate of hash runs

### ISSUE THREAD

**Issue by contentfree** [dry-rb/dry-validation#227]
I found today that given a rule `required(:name).filled(:str?, size?:(1..100), unique?: {scope: :company_id})` that the `unique` custom predicate didn't run. Changing the declaration to `required(:name).filled(:str?, unique?: {scope: :company_id}, size?:(1..100))` made the `unique` one run – but then the size one didn't.

Converting to the form `required(:name).filled { size?(1..100) & unique?(scope: :company_id) }` fixes it.

**Comment by solnic** [comment]
Could you show me an example? These predicates are being ANDed so I bet you passed either an empty value or a non-string value. Although I'm not sure why the block-style definition would fix it.

**Comment by contentfree** [comment]
I can't put the full example, but…

Given a custom predicate with definition: `def unique?(attribute, options, value)` in a custom schema (`Fiddleback::Validations::Schema`), I have a validation block:

```ruby
validation schema: FiddlebackManager::Validations::Schema do
  required(:name).filled(size?: (1..160), unique?: [:name, scope: :company_id])
  # ... other validations
end
```

The database has a unique index on `(company_id, name)`. 

Running tests for that validation result in an ActiveRecord::RecordNotUnique exception – very unexpected for it to not be caught & handled by dry-v. Changing the validation block to the following results in passing tests:

```ruby
validation schema: FiddlebackManager::Validations::Schema do
  required(:name).filled{ size?(1..160) & unique?(:name, scope: :company_id) }
  # ... other validations
end
```

**Comment by solnic** [comment]
This is a silly bug. I'll fix and push 0.9.4.

**Comment by solnic** [comment]
btw one small tip: don't use `filled` when you specify `size?` because `filled` checks if the size is greater than 0, so it's redundant in your case. In the future I plan to add something that will analyze rule tree and warn you when redundant rules have been defined :)
